prow/plugins/plugins.go

```diff
@@ -459,6 +460,18 @@ type Welcome struct {
 	MessageTemplate string `json:"message_template,omitempty"`
 }
 
+// CherryPickUnapproved is the config for the cherrypick-unapproved plugin.
+type CherryPickUnapproved struct {
+	// BranchRegexp is the regular expression for branch names such that
+	// the plugin treats only PRs against these branch names as cherrypick PRs.
+	// Compiles into BranchRe during config load.
+	BranchRegexp string         `json:"branchregexp,omitempty"`
+	BranchRe     *regexp.Regexp `json:"-"`
+	// Comment is the comment added by the plugin while adding the
+	// `do-not-merge/cherry-pick-not-approved` label.
+	Comment string `json:"comment,omitempty"`
```

Comment:
This isn't actually the comment that is added. It is a format string for the comment. I think we should use a literal value here instead. The parameters to the format string are not exposed to the user so it isn't clear how to construct the comment format string. We also don't validate that the provided string is valid.

Reply:
Yes, having the user provide a format string that must use parameters that aren't exposed to the user isn't good UX. They would need to look at the code to see how the format string is actually used.
We either need to have the user provide a literal string (not a format string) or we need to clearly specify what the format string parameters are and in what order they must appear so that the user can construct their own format string without looking at the code.

Reply:
Ah of course! Fixed to use a literal value. :+1: